test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -128,15 +133,50 @@ def _test_common(
                 )
 
     def _test_code_common(
-        self, mod, inputs, include_ops, exclude_ops, atol=1e-5, rtol=1.3e-6
+        self,
+        mod,
+        inputs,
+        include_ops,
+        exclude_ops,
+        atol=1e-5,
+        rtol=1.3e-6,
+        check_quantization=False,
+        check_dynamic=False,
     ):
-        with torch.no_grad():
-            clone_inputs = self._clone_inputs(inputs)
-            expected = mod(*inputs)
-            actual, (source_code,) = run_and_get_code(
-                torch.compile(mod, fullgraph=True), *clone_inputs
+        def _check_has_dynamic_shape(source_code):
+            # Ref to https://github.com/pytorch/pytorch/blob/3fe8417643c8d6c2b3d95552cd90321d141b5d54
+            # /test/inductor/test_torchinductor_codegen_dynamic_shapes.py#L82-L95
+            for_loop_found = False
+            has_dynamic = False
+            lines = source_code.split("\n")
+            for line in lines:
+                if "for(" in line:
+                    for_loop_found = True
+                    if re.search(r";.*ks.*;", line) is not None:
+                        has_dynamic = True
+                        break
+            self.assertTrue(
+                has_dynamic,
+                msg=f"Failed to find dynamic for loop variable\n{source_code}",
             )
-            torch.testing.assert_close(actual, expected, atol=atol, rtol=rtol)
+            self.assertTrue(for_loop_found, f"Failed to find for loop\n{source_code}")
```

Comment:
Thanks for the comment. Move this `_check_has_dynamic_shape` into `pytorch/torch/testing/_internal/inductor_utils.py`. So, it can be shared for usage.